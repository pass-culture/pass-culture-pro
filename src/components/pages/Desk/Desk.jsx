import PropTypes from 'prop-types'
import React, { Component } from 'react'

import AppLayout from 'app/AppLayout'
import TextInput from 'components/layout/inputs/TextInput/TextInput'
import PageTitle from 'components/layout/PageTitle/PageTitle'
import Titles from 'components/layout/Titles/Titles'
import { formatLocalTimeDateString } from 'utils/timezone'

class Desk extends Component {
  constructor(props) {
    super(props)

    this.state = {
      booking: null,
      isDisabledButton: true,
      isUsedToken: false,
      level: '',
      message: 'Saisissez une contremarque',
      token: '',
    }

    this.TOKEN_MAX_LENGTH = 6
    this.VALID_TOKEN_SYNTAX = /[^a-z0-9]/i
    this.BOOKING_ALREADY_USED = 410
  }

  formattedBookingDate = booking =>
    !booking.datetime
      ? 'Permanent'
      : formatLocalTimeDateString(
        booking.datetime,
        booking.venueDepartementCode,
        "dd/MM/yyyy - HH'h'mm"
      )

  firstErrorMessageFromApi = body => Object.keys(body)[0]

  validateToken = event => {
    const { getBooking } = this.props
    const token = event.target.value.toUpperCase()
    const { canCheckTheToken, level, message } = this.getStatusFromToken(token)
    this.setState({
      booking: null,
      isDisabledButton: true,
      isUsedToken: false,
      level,
      message,
      token,
    })

    if (canCheckTheToken) {
      getBooking(token)
        .then(booking => {
          this.setState({
            booking,
            isDisabledButton: false,
            message: 'Coupon vérifié, cliquez sur "Valider" pour enregistrer',
          })
        })
        .catch(error => {
          if (error.status === this.BOOKING_ALREADY_USED) {
            error.json().then(body => {
              this.setState({
                level: '',
                isUsedToken: true,
                message: body[this.firstErrorMessageFromApi(body)],
              })
            })
          } else {
            error.json().then(body => {
              this.setState({
                level: 'error',
                message: body[this.firstErrorMessageFromApi(body)],
              })
            })
          }
        })
    }
  }

  getStatusFromToken = token => {
    if (token === '') {
      return {
        canCheckTheToken: false,
        level: '',
        message: 'Saisissez une contremarque',
      }
    }

    if (token.match(this.VALID_TOKEN_SYNTAX) !== null) {
      return {
        canCheckTheToken: false,
        level: 'error',
        message: 'Caractères valides : de A à Z et de 0 à 9',
      }
    }

    if (token.length < this.TOKEN_MAX_LENGTH) {
      return {
        canCheckTheToken: false,
        level: '',
        message: `Caractères restants : ${this.TOKEN_MAX_LENGTH - token.length}/${
          this.TOKEN_MAX_LENGTH
        }`,
      }
    }

    return {
      canCheckTheToken: true,
      level: '',
      message: 'Vérification...',
    }
  }

  registrationOfToken = token => event => {
    event.preventDefault()
    const { validateBooking } = this.props
    this.setState({
      message: 'Validation en cours...',
    })

    validateBooking(token)
      .then(() => {
        const { trackValidateBookingSuccess } = this.props
        this.setState({
          message: 'Contremarque validée !',
        })
        trackValidateBookingSuccess(token)
      })
      .then(() => {
        this.setState({
          isDisabledButton: false,
          isUsedToken: true,
        })
      })
      .catch(error => {
        error.json().then(body => {
          this.setState({
            level: 'error',
            message: body[this.firstErrorMessageFromApi(body)],
          })
        })
      })
  }

  invalidationOfToken = token => event => {
    event.preventDefault()
    const { invalidateBooking } = this.props
    this.setState({
      message: 'Invalidation en cours...',
    })

    invalidateBooking(token)
      .then(() => {
        this.setState({
          message: 'Contremarque invalidée !',
        })
      })
      .then(() => {
        this.setState({
          isUsedToken: false,
          isDisabledButton: false,
        })
      })
  }

  render() {
    const { booking, isDisabledButton, isUsedToken, level, message, token } = this.state

    return (
      <AppLayout layoutConfig={{ pageName: 'desk' }}>
        <PageTitle title="Guichet" />
        <Titles title="Guichet" />
        <p className="advice">
          {
            'Saisissez les contremarques présentées par les jeunes afin de les valider ou de les invalider.'
          }
        </p>
        <form>
          <TextInput
            label="Contremarque"
            maxLength={this.TOKEN_MAX_LENGTH}
            name="token"
            onChange={this.validateToken}
            placeholder="ex : AZE123"
            type="text"
            value={token}
          />

          {booking && (
            <div
              aria-live="polite"
              aria-relevant="all"
              className="booking-summary"
            >
              <div>
                <div className="desk-label">
                  {'Utilisateur : '}
                </div>
                <div className="desk-value">
                  {booking.userName}
                </div>
              </div>
              <div>
                <div className="desk-label">
                  {'Offre : '}
                </div>
                <div className="desk-value">
                  {booking.offerName}
                </div>
              </div>
              <div>
                <div className="desk-label">
                  {'Date de l’offre : '}
                </div>
                <div className="desk-value">
                  {this.formattedBookingDate(booking)}
                </div>
              </div>
              <div>
                <div className="desk-label">
                  {'Prix : '}
                </div>
                <div className="desk-value">
                  {`${booking.price} €`}
                </div>
              </div>
            </div>
          )}

          <div className="desk-button">
            {isUsedToken && (
              <button
                className="secondary-button"
                onClick={this.invalidationOfToken(token)}
                type="submit"
              >
                {'Invalider la contremarque'}
              </button>
            )}
            {!isUsedToken && (
              <button
                className="primary-button"
                disabled={isDisabledButton}
                onClick={this.registrationOfToken(token)}
                type="submit"
              >
                {'Valider la contremarque'}
              </button>
            )}
          </div>

          <div
            aria-live="assertive"
            aria-relevant="all"
            className={`desk-message ${level}`}
          >
            {message}
          </div>
        </form>
      </AppLayout>
    )
  }
}

Desk.propTypes = {
  getBooking: PropTypes.func.isRequired,
  invalidateBooking: PropTypes.func.isRequired,
  trackValidateBookingSuccess: PropTypes.func.isRequired,
  validateBooking: PropTypes.func.isRequired,
}

export default Desk
