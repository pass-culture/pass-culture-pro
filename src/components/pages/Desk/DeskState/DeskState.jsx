import { Icon } from 'pass-culture-shared'
import React from 'react'
import PropTypes from 'prop-types'
import { formatLocalTimeDateString } from '../../../../utils/timezone'

const displayBookingDate = booking => {
  if (!booking) {
    return null
  }

  if (booking.date) {
    return formatLocalTimeDateString(booking.date, booking.venueDepartementCode)
  }

  return 'Permanent'
}

const DeskState = ({ message, level, booking }) => (
  <div className="desk-state">
    <table className="booking-summary">
      <tbody>
        <tr>
          <th>Utilisateur :</th>
          <td>{booking && booking.userName}</td>
        </tr>
        <tr>
          <th>Offre :</th>
          <td>{booking && booking.offerName}</td>
        </tr>
        <tr>
          <th>Date de l'offre :</th>
          <td>{displayBookingDate(booking)}</td>
        </tr>
      </tbody>
    </table>
    <div className={`state ${level}`}>
      {level === 'success' && <Icon svg="picto-validation" />}
      {level === 'error' && <Icon svg="picto-echec" />}
      <span>{message}</span>
    </div>
  </div>
)

DeskState.propTypes = {
  booking: PropTypes.shape({
    date: PropTypes.string,
    offerName: PropTypes.string,
    userName: PropTypes.string,
    venueDepartementCode: PropTypes.string,
  }),
  message: PropTypes.string.isRequired,
  level: PropTypes.string.isRequired,
}

export default DeskState
