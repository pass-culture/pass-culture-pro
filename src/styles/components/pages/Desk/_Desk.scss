.desk-page {
  .form {
    text-align: center;

    .columns {
      width: 392px;
      margin: auto;
    }

    input[type='text'] {
      width: 316px;
      height: 72px;
      font-size: 36px;
    }

    .buttonContainer {
      width: 92px;
      display: inline-block;
    }

    button {
      border-radius: 1.125em;
      font-size: 30px;
      height: 72px;
      margin-left: 20px;
    }

    .state {
      @include dotted($primary);

      margin: 20px 0;
      padding-top: 10px;
      display: flex;
      align-items: center;
      font-size: 20px;
      text-align: left;
      height: 60px;
      width: calc(100% - 110px);

      &.pending {
        color: green;
      }

      img {
        width: 50px;
      }

      span {
        padding-left: 20px;
        font-weight: 500;
      }
    }
  }

  .booking-summary {
    margin: 10px auto;
    margin-left: 28%;

    th {
      font-weight: normal;
      text-align: right;
    }

    td {
      padding-left: 5px;
      font-weight: bold;
    }
  }

  #exitlink {
    margin-top: -70px;
    float: right;
  }
}
