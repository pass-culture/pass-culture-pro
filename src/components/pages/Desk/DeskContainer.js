import { connect } from 'react-redux'
import { compose } from 'redux'

import { withTracking } from 'components/hocs'
import { fetchFromApiWithCredentials } from 'utils/fetch'

import Desk from './Desk'

export function mapStateToProps(state) {
  return {
    currentUser: state.users.currentUser,
  }
}

export const mapDispatchToProps = dispatch => ({
  getBooking: code =>
    fetchFromApiWithCredentials(`/v2/bookings/token/${code}`).then(booking => {
      dispatch({
        type: 'GET_DESK_BOOKINGS',
        payload: booking,
      })

      return booking
    }),
  validateBooking: code => fetchFromApiWithCredentials(`/v2/bookings/use/token/${code}`, 'PATCH'),
  invalidateBooking: code =>
    fetchFromApiWithCredentials(`/v2/bookings/keep/token/${code}`, 'PATCH'),
})

export const mergeProps = (stateProps, dispatchProps, ownProps) => ({
  ...stateProps,
  ...dispatchProps,
  trackValidateBookingSuccess: code => {
    ownProps.tracking.trackEvent({ action: 'validateBooking', name: code })
  },
})

export default compose(
  withTracking('Desk'),
  connect(mapStateToProps, mapDispatchToProps, mergeProps)
)(Desk)
