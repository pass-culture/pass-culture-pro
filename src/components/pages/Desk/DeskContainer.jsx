import Desk from './Desk'
import { compose } from 'redux'
import { withRedirectToSigninWhenNotAuthenticated } from '../../hocs/with-login'
import { connect } from 'react-redux'

export default compose(
  withRedirectToSigninWhenNotAuthenticated,
  connect()
)(Desk)
