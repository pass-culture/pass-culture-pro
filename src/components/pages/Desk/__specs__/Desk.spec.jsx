import { shallow } from 'enzyme'
import React from 'react'
import Desk from '../Desk'

import DeskState from '../DeskState/DeskState'

describe('src | components | pages | Desk | Desk ', () => {
  const options = {
    disableLifecycleMethods: true,
  }
  let getBookingFromCode
  let validateBooking
  let props
  let trackValidateBookingSuccess

  beforeEach(() => {
    getBookingFromCode = jest.fn()
    trackValidateBookingSuccess = jest.fn()
    validateBooking = jest.fn()

    props = {
      getBookingFromCode,
      trackValidateBookingSuccess,
      validateBooking,
    }
  })

  it('should render Desk component with default state', () => {
    // when
    const wrapper = shallow(<Desk {...props} />, options)

    // then
    expect(wrapper.state('booking')).toBeNull()
    expect(wrapper.state('code')).toBe('')
    expect(wrapper.state('status')).toBe('CODE_ENTER')
  })

  describe('render', () => {
    describe('desk status', () => {
      it('should render DeskState component with proper props when status is enter a code', () => {
        // when
        const wrapper = shallow(<Desk {...props} />, options)

        // then
        const deskState = wrapper.find(DeskState)
        expect(deskState).toHaveLength(1)
        expect(deskState.prop('message')).toBe('Saisissez un code')
        expect(deskState.prop('booking')).toBeNull()
        expect(deskState.prop('level')).toBe('pending')
      })

      it('should render DeskState component with proper props when status is typing a code', () => {
        // when
        const wrapper = shallow(<Desk {...props} />, options)
        wrapper.setState({ status: 'CODE_TYPING' })

        // then
        const deskState = wrapper.find(DeskState)
        expect(deskState).toHaveLength(1)
        expect(deskState.prop('message')).toBe('Caractères restants: 6/6')
        expect(deskState.prop('booking')).toBeNull()
        expect(deskState.prop('level')).toBe('pending')
      })

      it('should render DeskState component with proper props when status is invalid', () => {
        // when
        const wrapper = shallow(<Desk {...props} />, options)
        wrapper.setState({ status: 'CODE_SYNTAX_INVALID' })

        // then
        const deskState = wrapper.find(DeskState)
        expect(deskState).toHaveLength(1)
        expect(deskState.prop('message')).toBe('Caractères valides : de A à Z et de 0 à 9')
        expect(deskState.prop('booking')).toBeNull()
        expect(deskState.prop('level')).toBe('error')
      })

      it('should render DeskState component with proper props when status is verification', () => {
        // when
        const wrapper = shallow(<Desk {...props} />, options)
        wrapper.setState({ status: 'CODE_VERIFICATION_IN_PROGRESS' })

        // then
        const deskState = wrapper.find(DeskState)
        expect(deskState).toHaveLength(1)
        expect(deskState.prop('message')).toBe('Vérification...')
        expect(deskState.prop('booking')).toBeNull()
        expect(deskState.prop('level')).toBe('pending')
      })

      it('should render DeskState component with proper props when status is code already used', () => {
        // when
        const wrapper = shallow(<Desk {...props} />, options)
        wrapper.setState({ status: 'CODE_ALREADY_USED', booking: {} })

        // then
        const deskState = wrapper.find(DeskState)
        expect(deskState).toHaveLength(1)
        expect(deskState.prop('message')).toBe('Ce coupon est déjà enregistré')
        expect(deskState.prop('booking')).toStrictEqual({})
        expect(deskState.prop('level')).toBe('error')
      })

      it('should render DeskState component with proper props when status is code ok', () => {
        // when
        const wrapper = shallow(<Desk {...props} />, options)
        wrapper.setState({ status: 'CODE_VERIFICATION_SUCCESS', booking: {} })

        // then
        const deskState = wrapper.find(DeskState)
        expect(deskState).toHaveLength(1)
        expect(deskState.prop('message')).toBe(
          'Coupon vérifié, cliquez sur "Valider" pour enregistrer'
        )
        expect(deskState.prop('booking')).toStrictEqual({})
        expect(deskState.prop('level')).toBe('pending')
      })

      it('should render DeskState component with proper props when status is code registering', () => {
        // when
        const wrapper = shallow(<Desk {...props} />, options)
        wrapper.setState({
          status: 'CODE_REGISTERING_IN_PROGRESS',
          booking: {},
        })

        // then
        const deskState = wrapper.find(DeskState)
        expect(deskState).toHaveLength(1)
        expect(deskState.prop('message')).toBe('Enregistrement en cours...')
        expect(deskState.prop('booking')).toStrictEqual({})
        expect(deskState.prop('level')).toBe('pending')
      })

      it('should render DeskState component with proper props when status is code registered', () => {
        // when
        const wrapper = shallow(<Desk {...props} />, options)
        wrapper.setState({ status: 'CODE_REGISTERING_SUCCESS', booking: {} })

        // then
        const deskState = wrapper.find(DeskState)
        expect(deskState).toHaveLength(1)
        expect(deskState.prop('message')).toBe('Enregistrement réussi !')
        expect(deskState.prop('booking')).toStrictEqual({})
        expect(deskState.prop('level')).toBe('success')
      })

      it('should render DeskState component with proper props when status is code verification failed', () => {
        // when
        const wrapper = shallow(<Desk {...props} />, options)
        wrapper.setState({
          status: 'CODE_VERIFICATION_FAILED',
          booking: {},
          message: 'fake message',
        })

        // then
        const deskState = wrapper.find(DeskState)
        expect(deskState).toHaveLength(1)
        expect(deskState.prop('message')).toBe('fake message')
        expect(deskState.prop('booking')).toStrictEqual({})
        expect(deskState.prop('level')).toBe('error')
      })

      it('should render DeskState component with proper props when status is code registering failed', () => {
        // when
        const wrapper = shallow(<Desk {...props} />, options)
        wrapper.setState({
          status: 'CODE_REGISTERING_FAILED',
          booking: {},
          message: 'fake message',
        })

        // then
        const deskState = wrapper.find(DeskState)
        expect(deskState).toHaveLength(1)
        expect(deskState.prop('message')).toBe('fake message')
        expect(deskState.prop('booking')).toStrictEqual({})
        expect(deskState.prop('level')).toBe('error')
      })
    })

    describe('validate button', () => {
      it('should render a button which is disabled by default when code is not ok', () => {
        // when
        const wrapper = shallow(<Desk {...props} />, options)

        // then
        const button = wrapper.find('button')
        expect(button).toHaveLength(1)
        expect(button.prop('disabled')).toBe(true)
        expect(button.prop('className')).toBe('primary-button')
        expect(button.prop('type')).toBe('submit')
        expect(button.prop('onClick')).toStrictEqual(expect.any(Function))
      })

      it('should render a button which is enabled when code is ok', () => {
        // when
        const wrapper = shallow(<Desk {...props} />, options)
        wrapper.setState({ status: 'CODE_VERIFICATION_SUCCESS' })

        // then
        const button = wrapper.find('button')
        expect(button).toHaveLength(1)
        expect(button.prop('disabled')).toBe(false)
      })
    })
  })

  describe('functions', () => {
    describe('handleSuccessWhenGetBookingFromCode', () => {
      it('should update state status to code already used when a booking is retrieved and is validated', () => {
        // given
        const wrapper = shallow(<Desk {...props} />, options)
        const booking = {
          isValidated: true,
        }
        const state = {}
        const action = {
          payload: {
            datum: booking,
          },
        }

        // when
        wrapper.instance().handleSuccessWhenGetBookingFromCode(state, action)

        // then
        expect(wrapper.state('status')).toBe('CODE_ALREADY_USED')
        expect(wrapper.state('booking')).toBe(booking)
      })

      it('should update state status to code ok when a booking is not validated', () => {
        // given
        const wrapper = shallow(<Desk {...props} />, options)
        const booking = {
          isValidated: false,
        }
        const state = {}
        const action = {
          payload: {
            datum: booking,
          },
        }

        // when
        wrapper.instance().handleSuccessWhenGetBookingFromCode(state, action)

        // then
        expect(wrapper.state('status')).toBe('CODE_VERIFICATION_SUCCESS')
        expect(wrapper.state('booking')).toBe(booking)
      })
    })

    describe('handleFailWhenGetBookingFromCode', () => {
      it('should update state with code verification failed status and error message when booking retrieval failed', () => {
        // given
        const wrapper = shallow(<Desk {...props} />, options)
        const state = {}
        const action = {
          payload: {
            errors: [{ 1: 'error1' }, { 2: 'error2' }],
          },
        }

        // when
        wrapper.instance().handleFailWhenGetBookingFromCode(state, action)

        // then
        expect(wrapper.state('status')).toBe('CODE_VERIFICATION_FAILED')
        expect(wrapper.state('message')).toBe('error1 error2')
      })
    })

    describe('validateBooking', () => {
      it('should dispatch an action to validate booking using code', () => {
        // given
        const wrapper = shallow(<Desk {...props} />, options)

        // when
        wrapper.instance().handleCodeRegistration('ABCDEF')
        //wrapper.instance().handleSuccessWhenValidateBookin = jest.fn()

        // then
        expect(props.validateBooking).toHaveBeenCalledWith(
          'ABCDEF',
          undefined,
          expect.any(Function)
        )
      })
    })

    describe('handleSuccessWhenValidateBooking', () => {
      it('should update state status to code registered when code is registered successfully', () => {
        // given
        const wrapper = shallow(<Desk {...props} />, options)

        // when
        wrapper.instance().handleSuccessWhenValidateBooking()

        // then
        expect(wrapper.state('status')).toBe('CODE_REGISTERING_SUCCESS')
      })
    })

    describe('handleFailWhenValidateBooking', () => {
      it('should update state with code registering failed status and error message when code is not validated successfully', () => {
        // given
        const wrapper = shallow(<Desk {...props} />, options)
        const state = {}
        const action = {
          payload: {
            errors: [{ 1: 'error1' }, { 2: 'error2' }],
          },
        }

        // when
        wrapper.instance().handleFailWhenValidateBooking(state, action)

        // then
        expect(wrapper.state('status')).toBe('CODE_REGISTERING_FAILED')
        expect(wrapper.state('message')).toBe('error1 error2')
      })
    })

    describe('handleCodeChange', () => {
      it('should update status to code enter when code from input is empty', () => {
        // given
        const wrapper = shallow(<Desk {...props} />, options)
        const event = {
          target: {
            value: '',
          },
        }

        // when
        wrapper.instance().handleCodeChange(event)

        // then
        expect(wrapper.state('status')).toBe('CODE_ENTER')
      })

      it('should update status to code invalid when code from input is composed of invalid characters', () => {
        // given
        const wrapper = shallow(<Desk {...props} />, options)
        const event = {
          target: {
            value: 'ù^^``',
          },
        }

        // when
        wrapper.instance().handleCodeChange(event)

        // then
        expect(wrapper.state('status')).toBe('CODE_SYNTAX_INVALID')
      })

      it('should update status to code typing when code from input is lesser than 6 characters', () => {
        // given
        const wrapper = shallow(<Desk {...props} />, options)
        const event = {
          target: {
            value: 'ABCDE',
          },
        }

        // when
        wrapper.instance().handleCodeChange(event)

        // then
        expect(wrapper.state('status')).toBe('CODE_TYPING')
      })

      it('should update status to code verification when code respects format and number of characters', () => {
        // given
        const wrapper = shallow(<Desk {...props} />, options)
        const event = {
          target: {
            value: 'ABCDEF',
          },
        }

        // when
        wrapper.instance().handleCodeChange(event)

        // then
        expect(wrapper.state('status')).toBe('CODE_VERIFICATION_IN_PROGRESS')
      })

      it('should retrieve a booking information using code from input', () => {
        // given
        const wrapper = shallow(<Desk {...props} />, options)
        const event = {
          target: {
            value: 'ABCDEF',
          },
        }
        const input = wrapper.find('input')

        // when
        input.simulate('change', event)

        // then
        expect(wrapper.state('status')).toBe('CODE_VERIFICATION_IN_PROGRESS')
        expect(props.getBookingFromCode).toHaveBeenCalledWith(
          'ABCDEF',
          expect.any(Function),
          expect.any(Function)
        )
      })
    })
  })
})
