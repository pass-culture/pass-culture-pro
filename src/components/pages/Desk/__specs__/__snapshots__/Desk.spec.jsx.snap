// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | Desk | Desk  should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Desk
    getBookingFromCode={[MockFunction]}
    trackValidateBookingSuccess={[MockFunction]}
    validateBooking={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <HeroSection
          subtitle={null}
          title="Guichet"
        >
          <p
            className="subtitle"
          >
            Enregistrez les codes de réservations présentés par les porteurs du pass.
          </p>
        </HeroSection>,
        <div
          className="section form"
        >
          <p
            className="subtitle is-medium has-text-weight-bold"
          >
            Scannez un code-barres ou saisissez-le ci-dessous :
          </p>
          <input
            className="input is-undefined"
            maxLength={6}
            name="code"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="button"
            disabled={true}
            onClick={[Function]}
            type="submit"
          >
            Valider
          </button>
          <DeskState
            booking={null}
            level="pending"
            message="Saisissez un code"
          />
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="button is-primary is-medium is-pulled-right"
            id="exitlink"
            to="/accueil"
          >
            Terminer
          </NavLink>
        </div>,
      ],
      "name": "desk",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <p
            className="subtitle"
          >
            Enregistrez les codes de réservations présentés par les porteurs du pass.
          </p>,
          "subtitle": null,
          "title": "Guichet",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Enregistrez les codes de réservations présentés par les porteurs du pass.",
            "className": "subtitle",
          },
          "ref": null,
          "rendered": "Enregistrez les codes de réservations présentés par les porteurs du pass.",
          "type": "p",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="subtitle is-medium has-text-weight-bold"
            >
              Scannez un code-barres ou saisissez-le ci-dessous :
            </p>,
            <input
              className="input is-undefined"
              maxLength={6}
              name="code"
              onChange={[Function]}
              type="text"
              value=""
            />,
            <button
              className="button"
              disabled={true}
              onClick={[Function]}
              type="submit"
            >
              Valider
            </button>,
            <DeskState
              booking={null}
              level="pending"
              message="Saisissez un code"
            />,
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="button is-primary is-medium is-pulled-right"
              id="exitlink"
              to="/accueil"
            >
              Terminer
            </NavLink>,
          ],
          "className": "section form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Scannez un code-barres ou saisissez-le ci-dessous :",
              "className": "subtitle is-medium has-text-weight-bold",
            },
            "ref": null,
            "rendered": "Scannez un code-barres ou saisissez-le ci-dessous :",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input is-undefined",
              "maxLength": 6,
              "name": "code",
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": Object {
              "current": null,
            },
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Valider",
              "className": "button",
              "disabled": true,
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": "Valider",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "booking": null,
              "level": "pending",
              "message": "Saisissez un code",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Terminer",
              "className": "button is-primary is-medium is-pulled-right",
              "id": "exitlink",
              "to": "/accueil",
            },
            "ref": null,
            "rendered": "Terminer",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <HeroSection
            subtitle={null}
            title="Guichet"
          >
            <p
              className="subtitle"
            >
              Enregistrez les codes de réservations présentés par les porteurs du pass.
            </p>
          </HeroSection>,
          <div
            className="section form"
          >
            <p
              className="subtitle is-medium has-text-weight-bold"
            >
              Scannez un code-barres ou saisissez-le ci-dessous :
            </p>
            <input
              className="input is-undefined"
              maxLength={6}
              name="code"
              onChange={[Function]}
              type="text"
              value=""
            />
            <button
              className="button"
              disabled={true}
              onClick={[Function]}
              type="submit"
            >
              Valider
            </button>
            <DeskState
              booking={null}
              level="pending"
              message="Saisissez un code"
            />
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="button is-primary is-medium is-pulled-right"
              id="exitlink"
              to="/accueil"
            >
              Terminer
            </NavLink>
          </div>,
        ],
        "name": "desk",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <p
              className="subtitle"
            >
              Enregistrez les codes de réservations présentés par les porteurs du pass.
            </p>,
            "subtitle": null,
            "title": "Guichet",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Enregistrez les codes de réservations présentés par les porteurs du pass.",
              "className": "subtitle",
            },
            "ref": null,
            "rendered": "Enregistrez les codes de réservations présentés par les porteurs du pass.",
            "type": "p",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="subtitle is-medium has-text-weight-bold"
              >
                Scannez un code-barres ou saisissez-le ci-dessous :
              </p>,
              <input
                className="input is-undefined"
                maxLength={6}
                name="code"
                onChange={[Function]}
                type="text"
                value=""
              />,
              <button
                className="button"
                disabled={true}
                onClick={[Function]}
                type="submit"
              >
                Valider
              </button>,
              <DeskState
                booking={null}
                level="pending"
                message="Saisissez un code"
              />,
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="button is-primary is-medium is-pulled-right"
                id="exitlink"
                to="/accueil"
              >
                Terminer
              </NavLink>,
            ],
            "className": "section form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Scannez un code-barres ou saisissez-le ci-dessous :",
                "className": "subtitle is-medium has-text-weight-bold",
              },
              "ref": null,
              "rendered": "Scannez un code-barres ou saisissez-le ci-dessous :",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input is-undefined",
                "maxLength": 6,
                "name": "code",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": Object {
                "current": null,
              },
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Valider",
                "className": "button",
                "disabled": true,
                "onClick": [Function],
                "type": "submit",
              },
              "ref": null,
              "rendered": "Valider",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "booking": null,
                "level": "pending",
                "message": "Saisissez un code",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Terminer",
                "className": "button is-primary is-medium is-pulled-right",
                "id": "exitlink",
                "to": "/accueil",
              },
              "ref": null,
              "rendered": "Terminer",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
